vignettes/visualization-example.Rmd

```diff
@@ -0,0 +1,168 @@
+---
+title: "Example of interacting with and visualizing EFI standard data"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Data Access and Visualization}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, message=FALSE}
+library(EML)
+library(emld)
+library(lubridate)
+library(tibble)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+```
+
+
+Visualizing the simple, example forecast of population growth of two interacting species
+
+========================================================
+
+## Parsing metadata 
+
+
+First load the data using `EML`.
+
+```{r}
+eml_filename <- system.file("extdata", "forecast-eml.xml", package="EFIstandards")
+md <- read_eml(eml_filename)  #you could replace the variable here with a filename form a downloaded data package from a repository
+```
+
+Then to examine the details, use `EML::eml_get`. First, the title and abstract and who made it.
+
+```{r basic}
+eml_get(md, "title")
+eml_get(md, "abstract")
+eml_get(md, "creator")
+```
+
+Then diving in, the coverage and keywords
+
+```{r details}
+eml_get(md, "coverage")
+eml_get(md, "keywordSet")
+```
+
+Looks like it's a population level timeseries forecast of `r eml_get(md, "geographicDescription")[[1]]` for two species:
+
+```{r species}
+# these seem a bit of a pain to pull out
+spps <- eml_get(md, "taxonRankValue")
+sp1 <- paste(spps[1:2], collapse=" ")
+sp2 <- paste(spps[3:4], collapse=" ")
+sp1
+sp2
+```
+
+## Reading structured methods
+
+Load the additional metadata (see documentation of EFI forecasting standard):
+
+
+```{r methods}
+methods_md <- eml_get(md, "additionalMetadata")
+methods_md
+```
+
+This metadata contains details about the system forecast including number of state variables and parameters, and the methods used, including propagation of uncertainty. For example, details of the treatment of process error in this model are given by
+```{r process}
+methods_md %>% eml_get("process_error") # note could do this directly on \`md` object
+n_ensembles <- methods_md %>% eml_get("process_error") %>% eml_get("size")
+n_ensembles <- n_ensembles[[1]]
+```
+
+From this metadata, the output of this forecast should have error propagation via an ensemble of size `r n_ensembles`, two dimensions of process error (TODO one for each state variable? or two parameters govern it?) with no covariance between these. Further, one can check if the forecast used data assimilation (no):
```

Comment:
In this particular case the $\sigma$ for the two state variables happen to be the same.